test-fixtures/src/main/java/org/sonar/api/testfixtures/log/AbstractLogTester.java

```diff
@@ -48,19 +49,21 @@
     Level.ERROR, LoggerLevel.ERROR
   );
 
-  private final ListAppender<ILoggingEvent> listAppender = new ListAppender<>();
+  private final ConcurrentListAppender<ILoggingEvent> listAppender = new ConcurrentListAppender<>();
 
   protected void before() {
     getRootLogger().addAppender(listAppender);
     listAppender.start();
-    setLevel(LoggerLevel.INFO);
+    if (!levelChanged) {
+      setLevel(Level.INFO);
+    }
```

Comment:
Instead of storing the boolean to mark if the level was changed, what about storing the level it was changed to? This would eliminate the need to have boolean flag and if statement.

```
private Level level = Level.INFO
```

and then in the method:

```
setLevel(level)
```

Reply:
@marcin-majewski-sonarsource yes, that should work.
@henryju I'm not sure I get the difference between setting a level during the test, or setting the default level. It sounds like "setting the default level" is basically setting the level before the test runs?